kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigConfigMap.java

```diff
@@ -117,15 +111,63 @@ protected ConfigMap desired(KafkaProxy primary,
                     .addToLabels(standardLabels(primary))
                     .addNewOwnerReferenceLike(ResourcesUtil.newOwnerReferenceTo(primary)).endOwnerReference()
                 .endMetadata()
-                .withData(Map.of(CONFIG_YAML_KEY, generateProxyConfig(primary, context)))
+                .withData(data.build())
                 .build();
         // @formatter:on
     }
 
-    String generateProxyConfig(KafkaProxy primary,
-                               Context<KafkaProxy> context) {
-        ProxyModelBuilder proxyModelBuilder = ProxyModelBuilder.contextBuilder(context);
-        ProxyModel model = proxyModelBuilder.build(primary, context);
+    private static void addAcceptedConditions(Clock clock, ProxyModel proxyModel, ProxyConfigData data) {
+        var model = proxyModel.ingressModel();
+        for (ProxyIngressModel.VirtualClusterIngressModel virtualClusterIngressModel : model.clusters()) {
+            VirtualKafkaCluster cluster = virtualClusterIngressModel.cluster();
+            Condition removedResolvedRefs = ResourcesUtil.newConditionBuilder(clock, cluster)
+                    .withType(Condition.Type.ResolvedRefs)
+                    .build();
```

Comment:
The change to `RU#maybeAddOrUpdateCondition()` brings the semantic that adding a condition with a null status will remove any existing condition of the same type. When we build the list of conditions to add to the config map we're not ourselves calling `maybeAddOrUpdateCondition()`, but rather serializing a list to be use as the argument when the VKCRec picks up the list. This is why some of the expected resources in the `DerivedResourcesTest` look a little funky. 

I suppose it might be simpler to model these conditions being passed back to the VKCReconciler as an object, rather than a list:

```yaml
resolvedRefsCondition: ...
acceptedCondition: ...
```